### DISCUSSION TITLE
Работа с большой таблицей данных зависимых друг от друга

### DISCUSSION THREAD

**Discussion by blackcatru69** [totumonline/totum-ru-issues-and-discussions#3] [Как это сделать?] [answered]
Задача такая: заливается база автомобилей по столбцам (марка, модель, поколение, кузов, объем двигателя и так далее). 

Этот каталог нужен для того чтобы мастер приемщик быстро и правильно формировал карточку автомобиля клиента.

Далее создается простая таблица заказов с автомобилями клиентов, в которой присутствуют такие же столбцы.

Мастер приемщик добавляя автомобиль клиента начинает формировать его и селектами по порядку выбирает значения.

+ Первое — марка (тут присутствуют задвоение значений в селекте, но поиск работает нормально).

+ Далее — исходя из выбранной марки он должен выбрать модель в следующем столбце. Фильтр по марке и показать все модели которые в базе соответствуют значению этой марки.

+ Следующий столбец — поколение. Работает по принципу модели но с добавлением еще одного фильтра с сужением поиска).

+ И так далее пока не определится конкретный уникальный номер модели.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Спасибо Ярославу — мы на его кейсе записали небольшой комплексный пример с разбором не только каскадных селектов.

**https://youtu.be/Bn1P_qY3Ruo**

☝ таймкоды в описании к видео!

### Если говорить только о зависимых селектах

Например, если в зависимости от параметра надо брать отображения из разных таблиц:

```totum

=: if(condition: #condition_field = 1; then: $s1; else: $s2)

s1: selectRowListForSelect(table: 'table_1'; field: 'name')
s2: selectRowListForSelect(table: 'table_2'; field: 'name')

```
Или разный набор значений:

```totum

=: selectRowListForSelect(table: 'table'; field: 'name'; where: 'type' = #condition_field)

```

Для того, что бы это работало — надо включить параметр [индивидуальный расчет селекта для каждой строки](https://docs.totum.online/fields#fields-settings-codeSelectIndividual).

По умолчанию он выключен — это означает, что по умолчанию селект рассчитывается только для первой строки.

Когда параметр включен, для каждой строки селект будет рассчитан заново.

Этот же параметр обновляет зависимые селекты в хедере/футере/префильтре.

Поэтому — если у вас в коде селекта есть `if` или есть `where` зависящий от поля в таблице — то включайте индивидуальный расчет селекта для каждой строки.
